app/services/payment_builder.rb

```diff
@@ -0,0 +1,26 @@
+class PaymentBuilder
+  def initialize(application, expires_in_days)
+    @application = application
+    @expires_in_days = expires_in_days
+  end
+
+  def decide!
+    @application.create_payment(expires_at: expires_at) if part_payment?
+  end
+
+  private
+
+  def part_payment?
+    unless @application.payment?
+      @application.application_outcome == 'part' && evidence_check_payment_validation?
```

Comment:
PEDANTIC: there are two potential approaches here

a) one way of writing the above could be

```ruby
@application.application_outcome.equal?('part')
```

b) but a better way for me would be to take:

```ruby
@application.application_outcome == 'part' && evidence_check_payment_validation?
```
and move it to:

```ruby
def SOME_MEANINGFUL_NAME?
  @application.application_outcome.equal?('part') && evidence_check_payment_validation?
end
```

and then I'd re-write **part_payment?** as:

```ruby
def part_payment?
  SOME_MEANINGFUL_NAME? unless @application.payment?
end
```